statistics/statistics.go

```diff
@@ -241,27 +247,26 @@ func concatNameSpace(namespace []string) string {
 }
 
 // insertInToBuffer adds a new value into this' buffer object
-func (p *Plugin) insertInToBuffer(val interface{}, ns []string) {
-
-	if p.bufferCurSize == 0 {
+func (p *Plugin) insertInToBuffer(val interface{}, ns string) {
+	if p.bufferCurSize[ns] == 0 {
 		var buff = make([]interface{}, p.bufferMaxSize)
 		buff[0] = val
-		p.buffer[concatNameSpace(ns)] = buff
+		p.buffer[ns] = buff
 	} else {
-		p.buffer[concatNameSpace(ns)][p.bufferIndex] = val
+		p.buffer[ns][p.bufferIndex[ns]] = val
 	}
 }
 
 // updateCounters updates the meta informaiton (current size and index) of this' buffer object
-func (p *Plugin) updateCounters() {
-	if p.bufferCurSize < p.bufferMaxSize {
-		p.bufferCurSize++
+func (p *Plugin) updateCounters(ns string) {
+	if p.bufferCurSize[ns] < p.bufferMaxSize {
+		p.bufferCurSize[ns]++
 	}
 
-	if p.bufferIndex == p.bufferMaxSize-1 {
-		p.bufferIndex = 0
+	if p.bufferIndex[ns] == p.bufferMaxSize-1 {
+		p.bufferIndex[ns] = 0
 	} else {
-		p.bufferIndex++
+		p.bufferIndex[ns]++
```

Comment:
I don't understand your question. Are you asking why it needs to be a map or why we increment the index here?